### PR TITLE
Add December Pet Mod of the Season

### PR DESCRIPTION
**Pull request by Yoshiblaze**
Adds the new Pet Mod of the Season and Leader's Choice, being Little Colosseum and GSC Doubles, and removes the previous PMOTS and Leader's Choice.

GSC Doubles theoretically shouldn't need a mod folder but a couple mechanics had to be edited manually to give desired effects (being Encore making a Pokemon's attacks hit a random target and screens only reducing damage by 2/3), so it's needed for those.

### CODE REVIEW

**Comment by Yoshiblaze** [comment]
(Forgot to name the "Update formats-data.ts" commit - All it did was change tiering of everything into OU for the sake of challenge commands with new usage based tiers later on)

File: config/formats.ts [hunk 1]
```diff
@@ -356,35 +356,40 @@ export const Formats: import('../sim/dex-formats').FormatList = [
...
 	},
 	{
-		name: "[Gen 6] Megas Revisited",
-		desc: `A Gen 6 metagame where every Mega Evolution was reevaluated and redesigned.`,
-		mod: 'gen6megasrevisited',
-		ruleset: ['Standard', 'Swagger Clause', 'Mega Data Mod'],
-		banlist: ['Uber', 'Arena Trap', 'Shadow Tag', 'Soul Dew', 'Baton Pass', 'Blaziken + Speed Boost'],
+		name: "[Gen 9] Little Colosseum LC",
+		desc: `A Little Cup metagame that aims to buff weaker Pokemon and nerf LC Ubers to create a more diverse metagame.`,
+		mod: 'littlecolosseum',
```
Comment (KrisXV) [inline config/formats.ts:361]:
```suggestion
		mod: 'littlecolosseum',
		searchShow: false,
```

File: config/formats.ts [hunk 2]
```diff
@@ -356,35 +356,40 @@ export const Formats: import('../sim/dex-formats').FormatList = [
...
 		},
 		onSwitchIn(pokemon) {
 			this.add('-start', pokemon, 'typechange', (pokemon.illusion || pokemon).getTypes(true).join('/'), '[silent]');
 			pokemon.apparentType = pokemon.getTypes(true).join('/');
 		},
 	},
+	{
+		name: "[Gen 2] GSC Doubles",
+		desc: `A Gen 2 metagame that adds doubles to the game.`,
+		mod: 'gen2doubles',
```
Comment (KrisXV) [inline config/formats.ts:382]:
```suggestion
		mod: 'gen2doubles',
		searchShow: false,
```

File: config/formats.ts [hunk 3]
```diff
@@ -356,35 +356,42 @@ export const Formats: import('../sim/dex-formats').FormatList = [
...
-		name: "[Gen 6] Megas Revisited",
-		desc: `A Gen 6 metagame where every Mega Evolution was reevaluated and redesigned.`,
-		mod: 'gen6megasrevisited',
-		ruleset: ['Standard', 'Swagger Clause', 'Mega Data Mod'],
-		banlist: ['Uber', 'Arena Trap', 'Shadow Tag', 'Soul Dew', 'Baton Pass', 'Blaziken + Speed Boost'],
+		name: "[Gen 9] Little Colosseum LC",
+		desc: `A Little Cup metagame that aims to buff weaker Pokemon and nerf LC Ubers to create a more diverse metagame.`,
+		mod: 'littlecolosseum',
+		searchShow: false,
+		ruleset: ['Little Cup', 'Standard'],
```
Comment (KrisXV) [inline config/formats.ts:363]:
```suggestion
		ruleset: ['Little Cup', 'Standard'],
```

File: config/formats.ts [hunk 4]
```diff
@@ -356,35 +356,42 @@ export const Formats: import('../sim/dex-formats').FormatList = [
...
 			this.add('-start', pokemon, 'typechange', (pokemon.illusion || pokemon).getTypes(true).join('/'), '[silent]');
 			pokemon.apparentType = pokemon.getTypes(true).join('/');
 		},
 	},
+	{
+		name: "[Gen 2] GSC Doubles",
+		desc: `A Gen 2 metagame that adds doubles to the game.`,
+		mod: 'gen2doubles',
+		searchShow: false,
+		gameType: 'doubles',
```
Comment (KrisXV) [inline config/formats.ts:385]:
```suggestion
		gameType: 'doubles',
```

**Pull request merged**